internal_packages/worker-sync/lib/nylas-sync-worker-pool.coffee

```diff
@@ -118,6 +127,15 @@ class NylasSyncWorkerPool
 
     {create, modify, destroy}
 
+  _handleDeltaMetadata: (metadata) =>
+    for metadatum in metadata
+      klass = NylasAPI._apiObjectToClassMap[metadatum.object_type]
+      DatabaseStore.inTransaction (t) =>
+        t.find(klass, metadatum.object_id).then (model) ->
+          return Promise.resolve() unless model
```

Comment:
This will skip any metadata that's stored against objects which haven't yet come in through initial sync (but which were created prior to our first delta cursor). We should be able to deal with this when we do our initial metadata sync (which has to occur after regular initial sync).